propagation.go

```diff
@@ -0,0 +1,101 @@
+package opentracing
+
+///////////////////////////////////////////////////////////////////////////////
+// CORE PROPAGATION INTERFACES:
+///////////////////////////////////////////////////////////////////////////////
+
+// Injector is responsible for injecting Span instances in a manner suitable
+// for propagation via a format-specific "carrier" object. Typically the
+// injection will take place across an RPC boundary, but message queues and
+// other IPC mechanisms are also reasonable places to use a Injector.
+//
+// See Extractor and Span.Injector.
+type Injector interface {
+	InjectSpan(span Span, carrier interface{}) error
+}
+
+// Extractor is responsible for extracting Span instances from an
+// format-specific "carrier" object. Typically the extraction will take place
+// on the server side of an RPC boundary, but message queues and other IPC
+// mechanisms are also reasonable places to use a Extractor.
+//
+// See Injector and Tracer.Extractor.
+type Extractor interface {
+	JoinTrace(operationName string, carrier interface{}) (Span, error)
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// BUILTIN PROPAGATION FORMATS:
+///////////////////////////////////////////////////////////////////////////////
+
+// BuiltinFormat is used to demarcate the values within package `opentracing`
+// that are intended for use with the Tracer.Injector() and Tracer.Extractor()
+// methods.
+type BuiltinFormat byte
```

Comment:
@tschottdorf I went back to the enum, btw... it was nice to be able to refer to `BuiltinFormat` in comments about acceptable values for `format` params.